processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -163,7 +164,7 @@ public Object get()
       }
 
       @Override
-      public DimensionSelector makeDimensionSelector(final String dimension)
+      public DimensionSelector makeDimensionSelector(final String dimension, final ExtractionFn extractionFn)
```

Comment:
column selector is at the heart of lots of calls. Even though the lines added to the logic are very small, they will probably get called many many many times in potentially performance critical paths. If it weren't in such a critical portion I wouldn't be concerned. I'm most concerned that before and after the change for things like topN or groupBy do not change. DimExtractionTopN simply performs worse in general, but it would be good to know if this its performance changes much with the column selector changes.